core/src/shred_fetch_stage.rs

```diff
@@ -50,24 +50,25 @@ impl ShredFetchStage {
             .as_ref()
             .map(|(_, cluster_info)| cluster_info.keypair().clone());
 
-        // Only need root bank in order to check feature statuses later on;
-        // can demote to only fetching root slot once those features go away.
-        let (mut root_bank, mut last_slot) = {
+        let (mut last_root, mut last_slot, mut slots_per_epoch) = {
             let bank_forks_r = bank_forks.read().unwrap();
-            (bank_forks_r.root_bank(), bank_forks_r.highest_slot())
+            let root_bank = bank_forks_r.root_bank();
+            (
+                root_bank.slot(),
+                root_bank.get_slots_in_epoch(root_bank.epoch()),
+                bank_forks_r.highest_slot(),
```

Comment:
Seems like here the order of `last_slot` and `slots_per_epoch` is reveresed, so they are being assigned wrong values.

Reply:
patched in https://github.com/solana-labs/solana/pull/34338

Reply:
Ahh, good catch